ocw/lib/db.py

```diff
@@ -222,6 +229,48 @@ def update_run():
         init_cron()
 
 
+def is_delete_instance_allowed(instance):
+    return 'openqa_created_by' in instance.csp_info
+
+
+def delete_instance(instance):
+    if not is_delete_instance_allowed(instance):
+        raise PermissionError('This instance isn\'t managed by openqa')
+
+    logger.debug("[{}][{}] Delete instance {}".format(
+                instance.provider, instance.vault_namespace, instance.instance_id))
+    if (instance.provider == ProviderChoice.AZURE):
+        Azure(instance.vault_namespace).delete_resource(instance.instance_id)
+    elif (instance.provider == ProviderChoice.EC2):
+        EC2(instance.vault_namespace).delete_instance(instance.instance_id)
+    elif (instance.provider == ProviderChoice.GCE):
+        GCE(instance.vault_namespace).delete_instance(instance.instance_id, instance.region)
+    else:
+        raise NotImplementedError(
+                "Provider({}).delete() isn't implementd".format(instance.provider))
+
+    instance.state = StateChoice.DELETING
+    instance.save()
+
+
+def auto_delete_instances():
+    cfg = ConfigFile()
+    for vault_namespace in cfg.getList(['vault', 'namespaces'], ['']):
+        o = Instance.objects
+        o = o.filter(state=StateChoice.ACTIVE, vault_namespace=vault_namespace,
+                     max_age__gt=timedelta(0), age__gte=F('max_age'))
+        for i in o:
+            if not is_delete_instance_allowed(i):
```

Comment:
```suggestion
            if is_delete_instance_allowed(i):
```
positive branches always easier to read also you can skip one line :D

Reply:
done